src/geometry.cpp

```diff
@@ -536,6 +536,196 @@ extern "C" double openmc_calculate_optical_thickness(
+
+    total_sigma_t += p.macro_xs().total;
+    total_sigma_a += p.macro_xs().absorption;
+    valid_samples++;
+  }
+
+  if (valid_samples > 0) {
+    *sigma_t_output = total_sigma_t / valid_samples;
+    *sigma_a_output = total_sigma_a / valid_samples;
+  } else {
+    *sigma_t_output = 0.0;
+    *sigma_a_output = 0.0;
+  }
+}
+
+extern "C" void openmc_get_transport_operator_between_voxels(
+  Position start_voxel_min, Position start_voxel_max, Position end_voxel_min,
+  Position end_voxel_max, int num_rays, double* transport_operator_output)
+{
+  double total_transport = 0.0;
+  SpatialBox start_box(start_voxel_min, start_voxel_max);
+  SpatialBox end_box(end_voxel_min, end_voxel_max);
+  int64_t id = 1;
+  uint64_t seed = init_seed(id, STREAM_SOURCE);
+  int valid_samples = 0;
+
+#pragma omp parallel for reduction(+ : total_transport, valid_samples)
+
+  for (int i = 0; i < num_rays; ++i) {
+    Position start_sampled_position = start_box.sample(&seed);
+    Position end_sampled_position = end_box.sample(&seed);
+
+    double distance = (end_sampled_position - start_sampled_position).norm();
+    double optical_thickness = openmc_calculate_optical_thickness(
+      start_sampled_position, end_sampled_position);
+
+    double transport = std::exp(-optical_thickness) / (distance * distance);
+    total_transport += transport;
+    valid_samples++;
+  }
+
+  if (valid_samples > 0) {
+    *transport_operator_output = total_transport / valid_samples;
+  } else {
+    *transport_operator_output = 0.0;
+  }
+}
+
+extern "C" void openmc_get_self_transport_operator(Position voxel_min,
+  Position voxel_max, int num_rays, double* self_transport_output)
+{
+  double total_transport = 0.0;
+  SpatialBox voxel_box(voxel_min, voxel_max);
+  int64_t id = 1;
+  uint64_t seed = init_seed(id, STREAM_SOURCE);
+  int valid_samples = 0;
+
+  // Calculate rays per sample (num_rays / 6 rounded up)
+  // This is because we sample 6 directions for each ray
+  // and we want to ensure we have at least num_rays total samples
+  // across all directions.
+  int rays_per_direction = (num_rays + 5) / 6;
+  int total_rays = rays_per_direction * 6;
+
+#pragma omp parallel for reduction(+ : total_transport, valid_samples)
+  for (int i = 0; i < rays_per_direction; ++i) {
+    // Sample two random positions within the voxel
+    // r_j will be used to find b_1 through b_6
+    // the ray will be from r_i to b_1 through b_6
+    Position r_i = voxel_box.sample(&seed);
+    Position r_j = voxel_box.sample(&seed);
+
+    const Position directions[6] = {
+      {1.0, 0.0, 0.0},  // +x direction
+      {-1.0, 0.0, 0.0}, // -x direction
+      {0.0, 1.0, 0.0},  // +y direction
+      {0.0, -1.0, 0.0}, // -y direction
+      {0.0, 0.0, 1.0},  // +z direction
+      {0.0, 0.0, -1.0}  // -z direction
+    };
+
+    for (int dir = 0; dir < 6; dir++) {
+      Position b;
+
+      switch (dir) {
+      case 0: // +x direction
+        b = {voxel_max.x, r_j.y, r_j.z};
```

Comment:
[readability] assuming that `b` is initialized as `r_j`:
```suggestion
        b.x = voxel_max.x;
```
and so on for the other entries in the case statement.